### PR TITLE
MOD-2418 -TI.Map iOS: Support rotation on annotations

### PR DESCRIPTION
**Pull request by anil-shukla-axway**
https://jira.appcelerator.org/browse/MOD-2418

Added logic to rotate the custom annotations.

Note - Test on real device.

### CODE REVIEW

File: ios/Classes/TiMapView.m [hunk 1]
```diff
@@ -1192,6 +1202,13 @@ - (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotationProxy:(TiMap
     annView.clusteringIdentifier = [ann valueForUndefinedKey:@"clusterIdentifier"];
     annView.collisionMode = [TiUtils intValue:[ann valueForUndefinedKey:@"collisionMode"]];
     annView.displayPriority = [TiUtils floatValue:[ann valueForUndefinedKey:@"annotationDisplayPriority"] def:1000];
+    [UIView animateWithDuration:1.0
```
Comment (vijaysingh-axway) [inline ios/Classes/TiMapView.m:1205]: Why it is supported for iOS 11+, not for older iOS versions?

File: ios/Classes/TiMapView.m [hunk 2]
```diff
@@ -1245,6 +1262,54 @@ - (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotationProxy:(TiMap
   return annView;
 }
 
+- (void)updateLocation
+{
+
```
Comment (vijaysingh-axway) [inline ios/Classes/TiMapView.m:1267]: Please remove unnecessary line.

File: ios/Classes/TiMapView.m [hunk 3]
```diff
@@ -1245,6 +1262,54 @@ - (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotationProxy:(TiMap
...
+                       MKAnnotationView *annotationView = (MKAnnotationView *)[self.map viewForAnnotation:myAnnotation];
+                       annotationView.transform = CGAffineTransformMakeRotation(getAngle);
+                     }];
+  }
+}
+
+- (float)angleFromCoordinate:(CLLocationCoordinate2D)first
+                toCoordinate:(CLLocationCoordinate2D)second
+{
+
```
Comment (vijaysingh-axway) [inline ios/Classes/TiMapView.m:1298]: Please remove unnecessary line.

File: ios/Classes/TiMapView.m [hunk 4]
```diff
@@ -1245,6 +1262,54 @@ - (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotationProxy:(TiMap
...
+  MapAnnotation *myAnnotation;
+
+  CLLocationCoordinate2D oldLocation;
+  CLLocationCoordinate2D newLocation;
+
+  oldLocation.latitude = currentLocation.coordinate.latitude;
+  oldLocation.longitude = currentLocation.coordinate.longitude;
+  dispatch_queue_t serverDelaySimulationThread = dispatch_queue_create("com.xxx.serverDelay", nil);
+  dispatch_async(serverDelaySimulationThread, ^{
+    [NSThread sleepForTimeInterval:1.0];
```
Comment (vijaysingh-axway) [inline ios/Classes/TiMapView.m:1277]: Is it really required?

File: ios/Classes/TiMapView.m [hunk 5]
```diff
@@ -1245,6 +1262,54 @@ - (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotationProxy:(TiMap
...
+  });
+  newLocation.latitude = currentLocation.coordinate.latitude;
+  newLocation.longitude = currentLocation.coordinate.longitude;
+  float getAngle = [self angleFromCoordinate:oldLocation toCoordinate:newLocation];
+
+  //        NSLog(@"Index : %ld, Angle : %f, Latitude : %f, Longtitude : %f",(long)self.index, getAngle, newLocation.latitude, newLocation.longitude);
+  if (myAnnotation.coordinate.latitude) {
+
```
Comment (vijaysingh-axway) [inline ios/Classes/TiMapView.m:1285]: Remove unnecessary line.

File: ios/Classes/TiMapView.m [hunk 6]
```diff
@@ -1192,6 +1202,13 @@ - (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotationProxy:(TiMap
     annView.clusteringIdentifier = [ann valueForUndefinedKey:@"clusterIdentifier"];
     annView.collisionMode = [TiUtils intValue:[ann valueForUndefinedKey:@"collisionMode"]];
     annView.displayPriority = [TiUtils floatValue:[ann valueForUndefinedKey:@"annotationDisplayPriority"] def:1000];
+    [UIView animateWithDuration:1.0
+                          delay:0.0
+                        options:nil
+                     animations:^{
+                       annView.transform = CGAffineTransformRotate(annView.transform, 3.141593);
```
Comment (vijaysingh-axway) [inline ios/Classes/TiMapView.m:1209]: Instead of 3.141593 can we use M_PI

File: ios/MapAnnotation.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
+//
+//  MapAnnotation.h
+//  TestAnnotationMoving
+//
+//  Created by Nishith Shah on 14/11/16.
```
Comment (vijaysingh-axway) [inline ios/MapAnnotation.h:5]: Please replace commented lines with following. -
/**
 * Appcelerator Titanium Mobile
 * Copyright (c) 2009-Present by Appcelerator, Inc. All Rights Reserved.
 * Licensed under the terms of the Apache Public License
 * Please see the LICENSE included with this distribution for details.
 */

File: ios/MapAnnotation.m [hunk 8]
```diff
@@ -0,0 +1,13 @@
+//
+//  MapAnnotation.m
+//  TestAnnotationMoving
+//
+//  Created by Nishith Shah on 14/11/16.
```
Comment (vijaysingh-axway) [inline ios/MapAnnotation.m:5]: Same as above. See any other file in project.

File: ios/Classes/TiMapView.h [hunk 9]
```diff
@@ -50,6 +50,7 @@
 @property (nonatomic, readonly) CLLocationDegrees longitudeDelta;
 @property (nonatomic, readonly) CLLocationDegrees latitudeDelta;
 @property (nonatomic, readonly) NSArray *customAnnotations;
+@property (nonatomic, strong) NSMutableArray *arr;
```
Comment (vikas-goyal-axway) [inline ios/Classes/TiMapView.h:53]: ```arr``` name is not good please give it meaningful name

**Comment by vijaysingh-axway** [comment]
@anil-shukla-axway  Should this PR be closed as I can see you have open another PR #293 for same?

**Comment by anil-shukla-axway** [comment]
> @anil-shukla-axway Should this PR be closed as I can see you have open another PR #293 for same?

Yes, this can be closed as I have created the other PR which covers both the cases for MOD-2418/2450.

**Comment by vijaysingh-axway** [comment]
Closing in favor of PR #293

**Pull request closed without merging**